src/Compilers/Core/Portable/AssemblyUtilities.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Utilities
+{
+    internal static class AssemblyUtilities
+    {
+        /// <summary>
+        /// Given a path to an assembly, identifies files in the same directory
+        /// that could satisfy the assembly's dependencies. May throw.
+        /// </summary>
+        /// <remarks>
+        /// Dependencies are identified by simply checking the name of an assembly
+        /// reference against a file name; if they match the file is considered a
+        /// dependency. Other factors, such as version, culture, public key, etc., 
+        /// are not considered, and so the returned collection may include items that
+        /// cannot in fact satisfy the original assembly's dependencies.
+        /// </remarks>
+        /// <exception cref="IOException">If the file at <paramref name="filePath"/> does not exist or cannot be accessed.</exception>
+        /// <exception cref="BadImageFormatException">If the file is not an assembly or is somehow corrupted.</exception>
+        public static ImmutableArray<string> FindAssemblySet(string filePath)
+        {
+            Debug.Assert(filePath != null);
+            Debug.Assert(PathUtilities.IsAbsolute(filePath));
+
+            Queue<string> workList = new Queue<string>();
+            HashSet<string> assemblySet = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
`assemblySet` becomes the final set of items returned, and we don't want to end up with paths that differ only by case. `workList` just represents the paths we haven't yet checked for references; we never actually compare items in workList so a comparer isn't needed.